skellytracker/trackers/openpose_tracker/openpose_tracker.py

```diff
@@ -0,0 +1,129 @@
+
+import os
+
+
+class OpenPoseTracker(BaseCumulativeTracker):
+    def __init__(
+        self,
+        openpose_exe_path: Union[str, Path],
+        output_json_path: Union[str, Path],
+        net_resolution: str = "-1x640",
+        number_people_max: int = 1,
+    ):
+        """
+        Initialize the OpenPoseTracker.
+
+        :param recorder: An instance of OpenPoseRecorder for handling the output.
+        :param openpose_exe_path: Path to the OpenPose executable.
+        :param output_json_path: Directory where JSON files will be saved.
+        :param net_resolution: Network resolution for OpenPose processing.
+        :param number_people_max: Maximum number of people to detect.
+        """
+        # super().__init__(recorder=recorder)
+        self.openpose_exe_path = Path(openpose_exe_path)
+        self.output_json_path = Path(output_json_path)
+        self.net_resolution = net_resolution # TODO: this and num_people should be parameters for process_video, since we could use this one tracker to process videos with different parameters
+        self.number_people_max = number_people_max
+
+        super().__init__(
+            tracked_object_names=[],
+            recorder=OpenPoseRecorder(json_directory_path=output_json_path),
+        )
+
+        os.chdir(self.openpose_exe_path) # TODO: this can mess with things downstream if we're not sure to change the path back, we should just be explicit about the path throughout
+
+    def process_video(
+        self,
+        input_video_filepath: Union[str, Path],
+        output_video_filepath: Union[str, Path],
+        save_data_bool: bool = False,
+        use_tqdm: bool = True,
+        **kwargs,
+    ):
+        """
+        Run the OpenPose demo on a video file to generate JSON outputs
+        in a unique directory for each video.
+        """
+
+        # TODO: Add some check that the openpose executable exists (and maybe do a cross platform search like we do for blender in fmc?)
+
+        # Extract video name without extension to use as a unique folder name
+        video_name = Path(input_video_filepath).stem
+        unique_json_output_path = self.output_json_path / video_name
+        unique_json_output_path.mkdir(
+            parents=True, exist_ok=True
+        )  # Create the directory if it doesn't exist
+
+        # video_save_path = Path(output_video_filepath) / f"{video_name}_openpose.avi"
+
+        # Update the subprocess command to use the unique output directory
+        # TODO: subprocess call should probably be in a try/except with some form of error handling
+        subprocess.run(
+            [
+                "bin\OpenPoseDemo.exe",  # TODO: this is throwing the error `S607 Starting a process with a partial executable path`, needs to be something like usr/bin/OpenPoseDemo.exe, but I can't verify where it is (also does it need two backslashes to escape properly?)
```

Comment:
should this just be a parameter? We could implement something similar to the `find_blender_executable` function from freemocap to find the demo exe if the user doesn't provide it

Reply:
Let me do some testing on this one - there was some weirdness in openpose with how you need to have the exe path, but I need to refresh myself on what some of that was before saying anything concrete.

Reply:
I noticed we already take in an openpose exe path, so presumably we should just use that

Reply:
Okay, so basically - we could do the 'finding' function - but we don't want it to find the exe path, we would want it to find the 'root' OpenPose folder, which for example on my computer is `"C:\openpose"`. The reason for this is because the OpenPose demo only works if you change the working directory to the root openpose folder during runtime. 

So we want the root folder because that's what we need to change the working directory too. I suppose we could also try autofinding the exe file, and then taking the `.stem` of that path or something equivalent to find the root openpose folder from that path? 

However, I did just push an additional change, looking back at this code:

What we were previously doing is using `os.chdir(self.openpose_exe_path)` to change the current working directory to the root openpose folder (that parameter was poorly named on my part, the `openpose_exe_path` was in fact the path to `C:/openpose` or the equivalent, *not* the `exe`), and then running the subprocess call from there. However, using `os.chdir` would permanently change the working directly, which I'm sure would screw us over at some point.

What we can instead do is make a call within the subprocess that changes the working directory just for the duration of the subprocess, using `cwd = ...`. I made, tested, and pushed that change just now. (I also changed the `openpose_exe_path` to `openpose_root_folder_path` instead as a more accurate name).